### PR TITLE
Redirect new site message creation for no participants.

### PR DESCRIPTION
**Pull request by eschlange**
* Updated controller new logic.
* Added spec.

[#114007863]

### CODE REVIEW

File: app/controllers/think_feel_do_engine/coach/site_messages_controller.rb [hunk 1]
```diff
@@ -20,6 +20,9 @@ def show
 
       def new
         @participants = current_user.participants_for_group(@group)
+        redirect_to(coach_group_site_messages_path(@group),
+                    notice: "A group must have participants in order "\
```
Comment (ericcf) [inline app/controllers/think_feel_do_engine/coach/site_messages_controller.rb]: Should this be a `notice` or an `alert`?

Comment (eschlange) [inline app/controllers/think_feel_do_engine/coach/site_messages_controller.rb]: Ah yes, an alert makes more sense. Updating now.

File: spec/controllers/think_feel_do_engine/coach/site_messages_controller_spec.rb [hunk 2]
```diff
@@ -46,12 +46,21 @@ module Coach
           end
 
           it "responds with the group's participants but scoped to the current user" do
```
Comment (michaelwehrley) [inline spec/controllers/think_feel_do_engine/coach/site_messages_controller_spec.rb]: Should this test be titled something along the lines of that 'it responds with new template'?  This test actually has no idea if the participant is scoped to the `current_user` given the test setup.

Comment (eschlange) [inline spec/controllers/think_feel_do_engine/coach/site_messages_controller_spec.rb]: I agree, I'll update the description of this test.

**Comment by michaelwehrley** [comment]
:+1:

**Pull request merged**